### PR TITLE
AAD related changes in dataprotection aks ext CLI

### PR DESCRIPTION
**Pull request by anshulahuja98**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`. 

#### Manual tests performed
- Install extension without specifying anything related to AAD-> verified populated useAAD true and SA Uri.
- Install extension with useAAD = true ->  verified populated useAAD true and SA Uri.
- Install extension with useAAD = false ->  verified populated useAAD false and SA Uri.
- Update existing extension with useAAD and URI empty -> URI gets populated.
- Update current extension with no new params -> URI populated.

### CODE REVIEW

**Comment by anshulahuja98** [comment]
Draft currently, will raise after internal review and as per our required release timelines.
Merge ETA >1month

**Comment by anshulahuja98** [comment]
Discussed pipeline test failures with Bavneet, the team is aware of this bug and working on it.
Test failures are unrelated to my code changes

File: src/k8s-extension/azext_k8s_extension/partner_extensions/DataProtectionKubernetes.py [hunk 1]
```diff
@@ -128,12 +143,27 @@ def Update(
...
             bsl_specified = self.__is_bsl_specified(configuration_settings)
             self.__validate_and_map_config(configuration_settings, validate_bsl=bsl_specified)
             if bsl_specified:
                 self.__validate_backup_storage_account(cmd.cli_ctx, resource_group_name, cluster_name, configuration_settings)
 
+        # this step is for brownfield migrating to AAD
+        if configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_USE_AAD) is not None and configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_USE_AAD).lower() == "true":
+            logger.warning("useAAD flag is set to true. Please provide extension MSI Storage Blob Data Contributor role to the storage account.")
+
+        # SA Uri not provided in user input, and not populated in the original extension, we populate it.
```
Comment (Miraj50) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/DataProtectionKubernetes.py]: Consider the following case: 
Earlier BSL uri of SA1 was present. Now, an update comes, where only new BSL details of SA2 (bucket, sa, rg, sub) are provided. From what I can see, after the update, the BSL details will belong to SA2, but Uri will point to SA1, which will lead to failures.

File: src/k8s-extension/azext_k8s_extension/partner_extensions/DataProtectionKubernetes.py [hunk 2]
```diff
@@ -128,11 +143,26 @@ def Update(
...
+            logger.warning("useAAD flag is set to true. Please provide extension MSI Storage Blob Data Contributor role to the storage account.")
+
+        # SA details provided in user inputs, but did not provide SA URI.
+        if bsl_specified and configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI) is None:
+            logger.warning("storageAccountURI is not populated. Setting it to the storage account URI of provided storage account")
+            configuration_settings[self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI] = self.__get_storage_account_uri(cmd.cli_ctx, configuration_settings)
+            logger.warning(f"storageAccountURI: {configuration_settings[self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI]}")
+        # SA details not provided in user input, SA Uri not provided in user input, and also not populated in the original extension, we populate it.
+        elif not bsl_specified and original_extension.configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI) is None and configuration_settings.get(self.BACKUP_STORAGE_ACCOUNT_STORAGE_ACCOUNT_URI) is None:
```
Comment (Miraj50) [inline src/k8s-extension/azext_k8s_extension/partner_extensions/DataProtectionKubernetes.py]: If one just provides a BSL Uri for the current BSL, then I think it's not honored anymore.

**Comment by Miraj50** [comment]
Reviewed.

**Comment by anshulahuja98** [comment]
Final round of review and testing from Backup team is done
We require this PR To be merged around 15 January. 

For certain pipeline test failures, had pinged, @bavneetsingh16 that those are known issues and not regressed due to this PR

extension team will take up fix in early january.

Please let us know if there is any risk in merging/ releasing this around 15 January.

**Review by bavneetsingh16 — APPROVED** [review]

**Pull request merged**